### PR TITLE
Jakarta EE-fication of the articles MicroService

### PR DESCRIPTION
**Pull request by AdamBien**
The "articles" microservice was refactored and "optional" infrastructure code was removed. 
The focus was package structure and layering. Algorithms were not changed

### CODE REVIEW

File: articles-java-jee/src/main/java/com/ibm/cns/articles/control/DataAccessFacade.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.ibm.cns.articles.control;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+@ApplicationScoped
```
Comment (victorroeder) [inline articles-java-jee/src/main/java/com/ibm/cns/articles/control/DataAccessFacade.java]: Hi Adam,
I know it was not introduced by you but is @ApplicationScoped necessary?
BTW: Thx for your refactoring :-)

File: articles-java-jee/src/main/java/com/ibm/cns/articles/boundary/ArticlesService.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+@ApplicationScoped
```
Comment (victorroeder) [inline articles-java-jee/src/main/java/com/ibm/cns/articles/boundary/ArticlesService.java]: Hi Adam,
I know it was not introduced by you but is @ApplicationScoped necessary?
BTW: Thx for your refactoring :-)

File: articles-java-jee/src/main/java/com/ibm/cns/articles/control/JPADataAccess.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import com.ibm.cns.articles.entity.Article;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Resource;
+import javax.enterprise.context.ApplicationScoped;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.UserTransaction;
+
+@ApplicationScoped
```
Comment (victorroeder) [inline articles-java-jee/src/main/java/com/ibm/cns/articles/control/JPADataAccess.java]: Hi Adam,
I know it was not introduced by you but is @ApplicationScoped necessary?
BTW: Thx for your refactoring :-)

**Comment by nheidloff** [comment]
Hi @AdamBien, this looks pretty good.

For some reason 'getmultiple' only returns one article. I'll look into it asap.

**Comment by nheidloff** [comment]
Hi @AdamBien, I've looked into your PR some more. Everything works like a charm, when I use JPA against a SQL database. However the in-memory path doesn't work anymore since every article gets the id "0" assigned: https://github.com/AdamBien/cloud-native-starter/blob/master/articles-java-jee/src/main/java/com/ibm/cns/articles/entity/Article.java#L22 As result only one article is returned when invoking "getmultiple".

You know certainly better how to fix this. Just as an idea, does it make sense to use an in-memory database like H2 instead of the current implementation so that JPA id assignments work?

The default behaviour in our documentation is to use the in-memory storage, which for example we use in the workshops. So I'd like to get this path working again. Your advice is appreciated!

File: articles-java-jee/src/main/java/com/ibm/cns/articles/control/JPADataAccess.java (hunk 3 above)
Comment (AdamBien) [reply articles-java-jee/src/main/java/com/ibm/cns/articles/control/JPADataAccess.java]: I'm not done yet, it was the very first attempt. The ms is not JakartaEE-stic yet :-)

**Comment by nheidloff** [comment]
I've merged and added a fix to assign unique ids for articles in the in-memory store.

**Comment by nheidloff** [comment]
Hi @AdamBien, shouldn't the ArticlesService be under 'control' rather than 'boundary'?

**Pull request merged**